src/ucp/core/ucp_am.c

```diff
@@ -328,6 +328,12 @@ ucs_status_t ucp_worker_set_am_recv_handler(ucp_worker_h worker,
         return UCS_ERR_INVALID_PARAM;
     }
 
+    if (param->id > UINT16_MAX) {
+        ucs_error("AM handler id is too large (%u). "
```

Comment:
I would rephrase it like:
`AM handler id %u is outside of allowed range [0, %u], id, UINT16_MAX`
But up to you